### ISSUE TITLE
Neptune.ai logger slow, lags behind training 

### ISSUE THREAD

**Issue by Dunrar** [PyTorchLightning/pytorch-lightning#1261]
## 🐛 Bug

When running a script which trains multiple models after another, I came across the problem that the neptune.ai logger lags behind my training quite severely ( when the model has finished training, the logger is only about halfway there). This would not be such a big problem for me if the next model would start training while the logger still processes the logs of the previous model, but training continues only after the logger has finished. 

I tried logging fewer things, only logging metrics per epoch and disabling hardware monitoring by uninstalling psutil. I don't know if this is really a bug, but even if it is not, is there maybe a way to bypass/mitigate this? Is this maybe related to the windows timing error from #1186 @jakubczakon?  

### To Reproduce

1. Use the neptune.ai logger (with Windows) 
2. Log a metric 

### Expected behavior

The logger should keep up with training or continue logging previous models while another model is already being trained.

**Comment by jakubczakon** [comment]
Hi @Dunrar are you closing loggers per model explicitly:

```python
neptune_logger_1 = NeptuneLogger(close_after_fit=False)

# your logic

neptune_logger_1.experiment.stop()

neptune_logger_2 = NeptuneLogger(close_after_fit=False)

# your logic

neptune_logger_2.experiment.stop()
```

Could you elaborate on what do you mean by model here?

**Comment by Dunrar** [comment]
Yes, I tried that as well. By model I mean `model = MemoryTest(hparams)`. My code looks like this:

```
agg_losses = []
    for celltype in ['RNN', 'LSTM', 'GRU']:
        setattr(hparams, 'celltype', celltype)
        for layer in range(1, 4):
            setattr(hparams, 'n_layers', layer)
            for cell in range(1, 6):
                setattr(hparams, 'n_cells', cell)
                agg_losses_pred_col = []
                for predict_col in range(1, 6):
                    setattr(hparams, 'predict_col', predict_col)
                    print(' Updated hyperparameters: ', hparams)
                    agg_losses_runs = []
                    for run in range(10):
                        exp_name = 'Type ' + celltype + ', Layers ' + str(layer) + ', Cells ' + str(cell) + ', PredCol ' + str(predict_col) + ', Run ' + str(run)

                        neptune_logger = NeptuneLogger(
                            project_name="dunrar/bachelor-thesis",
                            experiment_name=exp_name,
                            params=vars(hparams),
                            close_after_fit=False
                            # tags=['truncated bptt', 'cyclical learning rates', 'epoch data reload', ...]
                        )
                        if hparams.early_stopping == 'yes':
                            early_stopping = EarlyStopping(
                                monitor='batch/mean_absolute_loss',
                                min_delta=hparams.min_delta,
                                patience=hparams.patience,
                                mode='min'
                            )
                        else:
                            early_stopping = False

                        if hparams.reload_data == 'yes':
                            reload_data = True
                        else:
                            reload_data = False

                        model = MemoryTest(hparams)
                        trainer = pl.Trainer(
                            benchmark=True,
                            checkpoint_callback=False,
                            logger=neptune_logger,
                            reload_dataloaders_every_epoch=reload_data,
                            gpus=hparams.cuda,
                            val_percent_check=0,
                            early_stop_callback=early_stopping,
                            default_save_path=src.settings.LOG_DIR,
                            max_epochs=hparams.epochs
                        )
                        trainer.fit(model)
                        neptune_logger.experiment.stop()
                        agg_losses_runs.append(model.final_loss)
                    agg_losses_pred_col.append(np.mean(agg_losses_runs))
                agg_losses.insert(0, agg_losses_pred_col)
```
But even when I only train one model and the model stops/finishes training (CPU load goes down as well) it takes a lot of time until the logger catches up and the console becomes responsive again.

**Comment by jakubczakon** [comment]
Got it, let me talk to the devs cause I think this should work nicely.

**Comment by pitercl** [comment]
Hi @Dunrar , I'm one of the developers of Neptune.ai.

The `stop()` method waits for some threads to join. One of those threads is responsible for sending channel values and stderr/out.

My current suspicion is that this problem is related to another problem with stderr/out we have on Windows. 

Can we check what happens when you do the following (upload_std*=False)?

```py
neptune_logger = NeptuneLogger(
                            project_name="dunrar/bachelor-thesis",
                            experiment_name=exp_name,
                            params=vars(hparams),
                            close_after_fit=False,
                            # tags=['truncated bptt', 'cyclical learning rates', 'epoch data reload', ...],
                            upload_stderr=False,
                            upload_stdout=False
                        )
```

Does the problem persist then?

**Comment by Dunrar** [comment]
Hey @pitercl , that's it, way faster now. It takes care of the other problem as well, of course.

**Comment by pitercl** [comment]
Hi @Dunrar, we're working on a fix. It should be release this week.

**Comment by pitercl** [comment]
Hi @Dunrar, I've released `neptune-client` in version `0.4.108`, which should fix the problem with colliding x-s in stdout/stderr. 

However, the problem with slow closing of experiments seems to lie elsewhere. I'll look into that as well and see what can be done.

Just wanted to give you a head's up on the progress.

**Comment by Dunrar** [comment]
Thank you @pitercl, I will check it out tomorrow! `upload_stderr=False` and `upload_stdout=False` already took care of the neptune logger lagging behind the training and it closes with just a short delay. Seems fine to me, I can't compare it to anything though.

**Comment by Borda** [comment]
@pitercl @jakubczakon will you send changes to Pl via PR or is it only Neptune task/issue?

**Comment by pitercl** [comment]
Hi @Borda, this is a Neptune-only task. I think this can be closed as it's not related to PL directly.
